### PR TITLE
fix #6071 feat(nimbus): Enable relative/absolute tab for tables + Highlights, cosmetic tweaks

### LINKED ISSUE #6071: Enable relative/absolute Results toggle for entire page
In EXP-1404, we're implementing a toggle for absolute/relative values in the Results Summary portion of the Results page.

Let's extend this functionality to include the entire page in case users want to see an absolute comparison for the other tables and because otherwise users may be confused at what comparison they're looking at in other parts of the page.

┆Issue is synchronized with this [Jiraserver Task](https://mozilla-hub.atlassian.net/browse/EXP-1551)

### PR DESCRIPTION
**Pull request by LZoog**
fixes #6071, but see [this comment](https://github.com/mozilla/experimenter/pull/6313#issuecomment-908799275) on why we're not extending for the "entire page" but just for tables that make sense

Because:
* Users would like to toggle between relative and absolute branch comparison values - functionality was previously implemented for two tables deemed most important, this change allows the toggle to switch in sets of tabs containing these tables plus, now, the highlights table

This commit:
* Enables the highlights table to toggle between absolute/relative comparison
* Removes the branch comparison button, uses Bootstrap's Tabs component instead for applicable tables since the toggle is not useful for the metric tables
* Contains cosmetic tweaks, including spacing updates, updating heading levels and adding a metrics heading, combining monitoring/analysis sections, and aligning tooltips horizontally

### CODE REVIEW

**Comment by LZoog** [comment]
I pushed an update. If you saw my first commit, I was originally passing `branchComparison` into Context, but it turns out we actually don't want the toggle functionality enabled for the entire page as the ticket describes because it's not useful for the metric tables (and an absolute comparison causes a bug in `ConfidenceInterval`) so we're not doing that any more. I was going down a performance rabbit hole with that approach anyway because the button toggle was super slow and memoizing was not doing what I wanted it to.

I decided to use `Tabs` similar to what we're doing on the home/directory page for the comparison toggle. Since it loads both versions on first render, there's no performance hiccup, and I thought it conveyed which tables were affected by the toggle in a nicer way than a floating button that only applied to certain tables.

👉 [Results page Storybook](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/27cb7fccc75cb473b975bbb93d3c33917f443553/nimbus-ui/index.html?path=/story/pages-results--basic)

File: app/experimenter/nimbus-ui/src/components/PageResults/index.test.tsx [hunk 1]
```diff
@@ -131,20 +133,16 @@ it("toggles between absolute and relative branch comparisons", async () => {
...
-    screen.queryByText("-0.46 to 0.51", { exact: false }),
-  ).not.toBeInTheDocument();
+  // shows relative by default
+  expect(relativeTab).toHaveAttribute("aria-selected", "true");
+  expect(absoluteTab).toHaveAttribute("aria-selected", "false");
+  fireEvent.click(absoluteTab);
 
-  fireEvent.click(screen.getByText("See relative comparison"));
-  await screen.findAllByText("-0.46 to 0.51", { exact: false });
-  expect(screen.queryByText("88.6%", { exact: false })).not.toBeInTheDocument();
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/PageResults/index.test.tsx:147]: I removed these since 1) the values are being tested individually in the other tables, and 2) this was giving me problems I think because the values are technically in the document as both sets of tables are rendered but one is hidden. `toBeVisible` wasn't being nice either 🤷‍♀️

File: app/experimenter/nimbus-ui/src/lib/visualization/constants.ts [hunk 2]
```diff
@@ -219,7 +220,7 @@ export const CONVERSION_METRIC_COLUMNS = [
 // display in the count metric table from left to right.
 export const COUNT_METRIC_COLUMNS = [
   {
-    name: "Count",
+    name: "Absolute Count",
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/lib/visualization/constants.ts:223]: I added the word "Absolute" for clarity. These metric tables show absolute and relative values - "Relative Improvement" is already stated above the confidence intervals.

File: app/experimenter/nimbus-ui/src/components/PageResults/TabbableTables/index.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        <TableResults {...{ experiment, branchComparison }} />
+      )}
+
+      {analysis.weekly && (
+        <TableResultsWeekly
+          metricsList={HIGHLIGHTS_METRICS_LIST}
+          {...{ branchComparison }}
+        />
+      )}
+    </div>
```
[the remark this answers is not in the document]
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/PageResults/TabbableTables/index.tsx:47]: Good call out! I don't think there is a case in which both of those can be falsey though. We have an `analysisAvailable` function that runs at the top of the page that checks for if at least one of those is truthy and if they're both falsey, the results are considered not available yet.

**Review by emtwo — DISMISSED** [review]
This looks great to me!

File: app/experimenter/nimbus-ui/src/lib/visualization/constants.ts [hunk 4]
```diff
@@ -202,9 +202,10 @@ export const CONVERSION_METRIC_COLUMNS = [
   {
     name: "Conversions / Total Users",
     displayType: DISPLAY_TYPE.CONVERSION_COUNT,
+    branchComparison: BRANCH_COMPARISON.ABSOLUTE,
   },
   {
-    name: "Conversion Rate",
+    name: "Absolute Conversion Rate",
```
[the remark this answers is not in the document]
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/lib/visualization/constants.ts:208]: Hmm, I added `branchComparison: BRANCH_COMPARISON.ABSOLUTE` since I changed `branchComparison` to be required for passing into `TableVisualizationRow` and I thought the [map at the top of that component](https://github.com/mozilla/experimenter/pull/6313/files#diff-59a511b5ef280cddb7a1aa227d06514c16304b43db64f9ca08df59eaaff702b8L31) referenced "absolute," but it looks like this is different than "primary metrics" so I'll look into this a bit further 👀

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Oh awesome @LZoog this is looking fantastic.  I see you've been taking some inspiration from github and it works wonderfully ;D

But I do have two pieces of feedback.

1. Not having the side borders on the individual metric tables looks out of place, could you add those in?

2. Using the tabs to switch between them is a great idea, but making the topmost one be a global relative/absolute gets confusing quickly when you move down the page, could you add the tabs to each individual metric so switching between them is interactable right on the metric? 🙏 

Screen Shot 2021-08-31 at 12 46 44 PM

**Comment by LZoog** [comment]
> Not having the side borders on the individual metric tables looks out of place, could you add those in?

Sure that's easy enough! Though, that is how we have had those up until this point.

> Using the tabs to switch between them is a great idea, but making the topmost one be a global relative/absolute gets confusing quickly when you move down the page, could you add the tabs to each individual metric so switching between them is interactable right on the metric? 🙏

I was actually thinking in the future, perhaps we could have the tabs on the left and have them scroll with the user, though I'd done a quick search and didn't see a Bootstrap solution easily applied for that, so this might be a better way to do it and account for users taking screenshots and wanting to see the name of the tab. I'll play around with that now and see if I can get it done quickly😬

File: app/experimenter/nimbus-ui/src/lib/visualization/constants.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/lib/visualization/constants.ts:208]: OK so, circling back here. We're not actually using `TableMetricConversion` on the Results page right now, but this is what it looks like in Storybook on `main`:

image

If I set the `branchComparison` here to `uplift`, we get this:
image

Then setting it to `absolute`, it's looking as it should:
image

I'll leave the branch comparison name to `absolute` since it renders the table as it did prior, but, I wonder if in your comment "It's conversions / total users" you were referring to the first column and not the second. I kind of feel like it should have the word "Absolute" as I changed it [here](https://github.com/mozilla/experimenter/pull/6313#discussion_r698903599)? But I can easily change it too, if we prefer "Conversion Rate". 👍

**Comment by LZoog** [comment]
@jaredlockhart I just pushed an update containing your requests - I'm going to see if I can add a story for `TableWithTabComparison` and maybe make some better use of types for the table passed in as a prop there, and I'd like to get confirmation from Marina on that other comment, but this is ready otherwise 😎

**Comment by emtwo** [comment]
Lauren, I'm so sorry I didn't notice this yesterday but the numbers for highlights absolute are not correct because they shouldn't be percentages. Here's a screenshot:

Screen Shot 2021-09-01 at 10 07 05 AM

E.g. the search for `msw_a` says `2062.5% to 2249.9% change`. But since it's absolute:
1) It should not be a percentage (i.e. it should not be multiplied by 100 and have a % sign)
2) It shouldn't say "change" since it's just absolute values.

Although, retention is by definition a % so that one should always have the % there even when it's absolute.
More generally, I noticed that the `%` sign isn't being correctly used (I'm so sorry I'm only noticing this now, I think I missed it in your last PR too)

Basically absolute values are counts so they should never have a `%` sign except for retention. Relative values should always have a `%` sign

File: app/experimenter/nimbus-ui/src/lib/visualization/constants.ts (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/src/lib/visualization/constants.ts:223]: Yes this makes all the difference in the world, helps me understand the page a lot better thank you 🙏

File: app/experimenter/nimbus-ui/src/lib/visualization/constants.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (LZoog) [reply app/experimenter/nimbus-ui/src/lib/visualization/constants.ts:208]: Marina gave the thumbs up on using the word "Absolute" here.

**Comment by LZoog** [comment]
Just pushed changes, hoping things look g2g. 😬

**Review by emtwo — APPROVED** [review]

**Review by jaredlockhart — APPROVED** [review]
Thank you for taking the time to wrap this up @LZoog !!! 🎉 🎉 🎉 🎉 🎉  Huge improvement to this page 👏 👏 👏 👏
